android-sdk/src/test/java/com/checkout/android_sdk/logging/FramesLoggingEventsTest.kt

```diff
@@ -1,36 +1,58 @@
 package com.checkout.android_sdk.logging
 
+import android.content.Context
+import androidx.test.core.app.ApplicationProvider
+import com.checkout.android_sdk.Utils.CheckoutTheme
 import com.checkout.android_sdk.Utils.Environment
+import com.checkout.android_sdk.Utils.filterNotNullValues
 import com.checkout.eventlogger.domain.model.MonitoringLevel
 import org.junit.Assert.assertEquals
+import org.junit.Before
 import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
 import java.util.Locale
 
+@RunWith(RobolectricTestRunner::class)
 class FramesLoggingEventsTest {
+    private lateinit var mockContext: Context
 
-  @Test
+
+    @Before
+    internal fun setUp() {
+        mockContext = ApplicationProvider.getApplicationContext()
+    }
+
+    @Test
     fun `test PaymentFormPresentedEvent map to correct values`() {
 
         assertEquals(getExpectedPaymentFormPresentedLoggingEvent().typeIdentifier,
-            FramesLoggingEventDataProvider.logPaymentFormPresentedEvent().typeIdentifier)
+            FramesLoggingEventDataProvider.logPaymentFormPresentedEvent(mockContext).typeIdentifier)
 
         assertEquals(getExpectedPaymentFormPresentedLoggingEvent().monitoringLevel,
-            FramesLoggingEventDataProvider.logPaymentFormPresentedEvent().monitoringLevel)
+            FramesLoggingEventDataProvider.logPaymentFormPresentedEvent(mockContext).monitoringLevel)
 
         assertEquals(getExpectedPaymentFormPresentedLoggingEvent().properties,
-            FramesLoggingEventDataProvider.logPaymentFormPresentedEvent().properties)
+            FramesLoggingEventDataProvider.logPaymentFormPresentedEvent(mockContext).properties)
 
     }
 
 
     private fun getExpectedPaymentFormPresentedLoggingEvent(): FramesLoggingEvent {
+        val checkoutTheme = CheckoutTheme(mockContext)
         val eventData = mapOf(
-            PaymentFormLanguageEventAttribute.locale to Locale.getDefault().toString()
+            PaymentFormLanguageEventAttribute.locale to Locale.getDefault().toString(),
+            PaymentFormLanguageEventAttribute.colorPrimary to checkoutTheme.getColorPrimaryProperty(),
+            PaymentFormLanguageEventAttribute.colorAccent to checkoutTheme.getColorAccentProperty(),
+            PaymentFormLanguageEventAttribute.colorButtonNormal to checkoutTheme.getColorButtonNormalProperty(),
+            PaymentFormLanguageEventAttribute.colorControlNormal to checkoutTheme.getColorButtonNormalProperty(),
+            PaymentFormLanguageEventAttribute.textColorPrimary to checkoutTheme.getTextColorPrimaryProperty(),
+            PaymentFormLanguageEventAttribute.colorControlActivated to checkoutTheme.getColorControlActivatedProperty(),
         )
         return FramesLoggingEvent(
             MonitoringLevel.INFO,
             FramesLoggingEventType.PAYMENT_FORM_PRESENTED,
-            properties = eventData
+            properties = eventData.filterNotNullValues()
```

Comment:
You should hardcode the expected values here and try to add some Null values here to add more test cases ( both null and not null to test behaviour ) instead of calling filterNotNullValues on eventData.